core/interfaces.go

```diff
@@ -151,3 +155,8 @@ type DNSResolver interface {
 	LookupCAA(string) ([]*dns.CAA, time.Duration, error)
 	LookupMX(string) ([]string, time.Duration, error)
 }
+
+// PublisherAuthority defines the public interface for the Boulder Publisher
+type PublisherAuthority interface {
```

Comment:
I would recommend just calling this interface (and implementation) Publisher. The "authority" part doesn't seem necessary, and creates a conflict for the PA acronym with the PolicyAuthority.